### PR TITLE
clean: add type hints

### PR DESCRIPTION
**Pull request by bolinocroustibat**
Small PR to add a few type hints, added on the go while working on some others features.

We can either create PRs to add types like this as we go, or use this same open draft PR to add some type hints continuously. Let me know what you prefer @magopian.

### CODE REVIEW

File: udata/utils.py [hunk 1]
```diff
@@ -175,7 +176,7 @@ def to_iso_date(dt):
         return "{dt.year:04d}-{dt.month:02d}-{dt.day:02d}".format(dt=dt)
 
 
-def to_iso_datetime(dt):
+def to_iso_datetime(dt: date | datetime) -> str:
```
Comment (magopian) [inline udata/utils.py:179]:
```suggestion
def to_iso_datetime(dt: date | datetime) -> str | None:
```

File: udata/utils.py [hunk 2]
```diff
@@ -302,7 +303,7 @@ class UnicodeLoremProvider(LoremProvider):
     word_list = [w + "é" for w in LoremProvider.word_list]
 
 
-def safe_unicode(string):
+def safe_unicode(string: str) -> str | None:
```
Comment (magopian) [inline udata/utils.py:306]: Should it be
```suggestion
def safe_unicode(string: bytes) -> str | None:
```
?

**Review by magopian — APPROVED** [review]
Nice work! Nice having more and more type hints.

A few nits and remarks, I'll let you decide if they make sense ;)

Regarding your question:

> We can either create PRs to add types like this as we go, or use this same open draft PR to add some type hints continuously.

The issue with adding more type hints continuously to this PR is that it'll have to stay around, unmerged, for a very long time, right? And it might become stale?

I'm afraid we don't really have a choice, and should create PRs as we go 👍

**Comment by magopian** [comment]
BTW, the test failure is because of me and #3112, if you get your branch up to date with master, the tests should all pass.

File: udata/core/reuse/search.py [hunk 3]
```diff
@@ -55,7 +55,7 @@ def mongo_search(cls, args):
         return reuses.order_by(sort).skip(offset).limit(args["page_size"]), reuses.count()
 
     @classmethod
-    def serialize(cls, reuse):
+    def serialize(cls, reuse: Reuse) -> dict:
         organization = None
         owner = None
         if reuse.organization:
```
[the remark this answers is not in the document]
Comment (bolinocroustibat) [reply udata/core/reuse/search.py:61]: Yes same here, but couldn't fix that - let's merge anyway and we'll fix more on the go later

File: udata/tags.py [hunk 4]
```diff
@@ -19,5 +19,5 @@ def normalize(value):
     return value
```
[the remark this answers is not in the document]
Comment (bolinocroustibat) [reply udata/tags.py:19]: I couldn't fin any solution for now so I merged anyway, to fix later!

**Pull request merged**